airflow/datadog_checks/airflow/airflow.py

```diff
@@ -27,22 +27,43 @@ def check(self, _):
 
         url_stable = self._url + "/api/v1/health"
         url_experimental = self._url + "/api/experimental/test"
+        url_stable_version = self._url + "/api/v1/version"
         can_connect_status = AgentCheck.OK
 
-        # Query the stable API first
-        resp = self._get_json(url_stable)
-        if resp is None:
+        # Choose which version of the API to use
+        if self._get_version(url_stable_version) is None:
             resp = self._get_json(url_experimental)
             if resp is None:
                 can_connect_status = AgentCheck.CRITICAL
             else:
                 self._submit_healthy_metrics_experimental(resp, tags)
         else:
-            self._submit_healthy_metrics_stable(resp, tags)
+            resp = self._get_json(url_stable)
+            if resp is None:
+                can_connect_status = AgentCheck.CRITICAL
+            else:
+                self._submit_healthy_metrics_stable(resp, tags)
 
         self.service_check('airflow.can_connect', can_connect_status, tags=tags)
         self.gauge('airflow.can_connect', int(can_connect_status == AgentCheck.OK), tags=tags)
 
+    def _get_version(self, url):
+        """Get version from stable API `/api/v1/version`"""
+        try:
+            resp = self.http.get(url)
+            resp.raise_for_status()
+            resp = resp.json()
+        except Exception as e:
+            self.log.debug("Couldn't collect version from URL: %s with exception: %s", url, e)
+        else:
+            try:
+                version = resp.get('version')
+            except AttributeError:
```

Comment:
Hmm, this is confusing. `get` is a method on a dictionary which doesn't throw any error for missing keys, but returns `None`. So you'll never get `AttributeError` here. **If** you switch to getting the key with the bracket notation, `resp['version']`, you'll get a `KeyError` for missing keys.

imho this checking logic should be in the caller method, this method should focus on getting the response payload.